kafka_consumer/datadog_checks/kafka_consumer/new_kafka_consumer.py

```diff
@@ -82,12 +89,37 @@ def check(self):
                 self._context_limit,
             )
 
+        self._save_broker_timestamps()
+
         # Report the metrics
         self._report_highwater_offsets(self._context_limit)
         self._report_consumer_offsets_and_lag(self._context_limit - len(self._highwater_offsets))
 
         self._collect_broker_metadata()
 
+    def _load_broker_timestamps(self):
+        """Loads broker timestamps from disk."""
+        self._broker_timestamps = defaultdict(dict)
+        try:
+            with open(self._broker_timestamps_filename, 'rb') as f:
```

Comment:
You should use the built in cache rather than manipulating files directly. Look at write_persistent_cache method on checks base